homeassistant/core.py

```diff
@@ -2282,6 +2287,7 @@ def async_set(
             # mypy does not understand this is only possible if old_state is not None
             old_last_reported = old_state.last_reported  # type: ignore[union-attr]
             old_state.last_reported = now  # type: ignore[union-attr]
+            old_state.last_reported_timestamp = timestamp  # type: ignore[union-attr]
```

Comment:
This was actually a bug, but it went unnoticed before because it worked the first time